src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            BulldozerProducer producer = new BulldozerProducer();
+            List<? extends Machine> machines = producer.get();
+            return new ArrayList<>(machines);
+        } else if (type.equals(Excavator.class)) {
+            ExcavatorProducer producer = new ExcavatorProducer();
+            List<? extends Machine> machines = producer.get();
+            return new ArrayList<>(machines);
+        } else if (type.equals(Truck.class)) {
+            TruckProducer producer = new TruckProducer();
+            List<? extends Machine> machines = producer.get();
```

Comment:
1. producers should be created once. make them as `private final` fields
2. redundant variables
```suggestion
            return new ArrayList<>(bulldozerProducer.get());
        } else if (type.equals(Excavator.class)) {
            return new ArrayList<>(excavatorProducer.get());
        } else if (type.equals(Truck.class)) {
```